src/libutil/compression.cc

```diff
@@ -13,12 +13,37 @@
 #include <brotli/encode.h>
 #endif // HAVE_BROTLI
 
+#if HAVE_ZSTD
+#include <zstd.h>
+// Not set by earlier versions, so be sure it's set
```

Comment:
Would it be better to do this check with autoconf, or by making a wrapper-header around `zstd.h`?
I think the answer depends on how many files you expect to do zstd stuff.

Reply:
Not sure, I'm not loving the macro usage either so suggestions welcome.  Not sure what you're suggesting in first clause-- the `HAVE_ZSTD` macro is the result of an autoconf check added in this PR.
The second suggestion, generating a wrapper, has potential re:making things less messy.

We could just make zstd non-optional (same for brotli) but isn't done for various reasons-- I think the most important being that not all distributions Nix wants to support provide these packages...

Suggestions welcome, I'll think about this a bit.  Popular alternative is to have build system selectively include files -- not sure if that's preferred or not.

Reply:
In the first alternative, I mean do a check in `configure.ac`:

```autoconf
AC_MSG_CHECKING([for ZSTD_CLEVEL_DEFAULT in zstd.h])
AC_PREPROC_IFELSE([AC_LANG_SOURCE([[#include <zstd.h>
#ifndef ZSTD_CLEVEL_DEFAULT
#error
#endif]])],[AC_MSG_RESULT([yes])], [AC_MSG_RESULT([no]), AC_DEFINE([ZSTD_CLEVEL_DEFAULT], [3])])
```

A wrapper header is probably better, that includes `<zstd.h>` and defines things that might not be present.